comp/core/tagger/impl-remote/remote.go

```diff
@@ -287,65 +286,51 @@ func (t *remoteTagger) GenerateContainerIDFromOriginInfo(originInfo origindetect
 }
 
 // queryContainerIDFromOriginInfo calls the local tagger to get the container ID from the Origin Info.
-func (t *remoteTagger) queryContainerIDFromOriginInfo(originInfo origindetection.OriginInfo) (containerID string, err error) {
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 50 * time.Millisecond
-	expBackoff.MaxInterval = 200 * time.Millisecond
-	expBackoff.MaxElapsedTime = 500 * time.Millisecond
-
-	err = backoff.Retry(func() error {
-		select {
-		case <-t.ctx.Done():
-			return &backoff.PermanentError{Err: errTaggerFailedGenerateContainerIDFromOriginInfo}
-		default:
-		}
-
-		// Fetch the auth token
-		if t.token == "" {
-			var authError error
-			t.token, authError = t.options.TokenFetcher()
-			if authError != nil {
-				_ = t.log.Errorf("unable to fetch auth token, will possibly retry: %s", authError)
-				return authError
-			}
+func (t *remoteTagger) queryContainerIDFromOriginInfo(originInfo origindetection.OriginInfo) (string, error) {
+	// Fetch the auth token
+	if t.token == "" {
+		var authError error
+		t.token, authError = t.options.TokenFetcher()
+		if authError != nil {
```

Comment:
here you can have race condition. if the function `queryContainerIDFromOriginInfo` is call by several goroutine. 

can we imagine to have the token retrieved during the remoteTagger initialisation?